### PR TITLE
Add message to Timeout failures

### PR DESCRIPTION
**Pull request by feedmeapples**
**What changed?**
Activity and Workflow timeout failures now contain message and source information

**Why?**
Such failures by default were showing up empty in web/tctl/any other consumer unless have a special case of processing internal Timeout structure

**How did you test it?**
Ran tests and tried creating a timeout failure and verified through web/tctl
image

**Potential risks**
No risks

### CODE REVIEW

File: common/failure/failure.go [hunk 1]
```diff
@@ -52,8 +58,10 @@ func NewResetWorkflowFailure(message string, lastHeartbeatDetails *commonpb.Payl
 	return f
 }
 
-func NewTimeoutFailure(timeoutType enumspb.TimeoutType) *failurepb.Failure {
+func NewTimeoutFailure(message string, timeoutType enumspb.TimeoutType) *failurepb.Failure {
 	f := &failurepb.Failure{
+		Message: fmt.Sprintf("%v: %v", enumspb.TimeoutType_name[int32(timeoutType)], message),
```
Comment (alexshtin) [inline common/failure/failure.go:63]: You need to swap arguments here:
```suggestion
		Message: fmt.Sprintf("%s: %v", message, timeoutType.String()]),
```

File: service/history/timerQueueActiveTaskExecutor.go [hunk 2]
```diff
@@ -219,7 +219,7 @@ Loop:
 			break Loop
 		}
 
-		timeoutFailure := failure.NewTimeoutFailure(timerSequenceID.timerType)
+		timeoutFailure := failure.NewTimeoutFailure("ActivityTask timeout", timerSequenceID.timerType)
```
Comment (alexshtin) [inline service/history/timerQueueActiveTaskExecutor.go:222]: Here is the thing: `TimeoutType` is not immutable and it is getting changed on line 238 bellow. Message should reflect it. I don't see good solution for it and I would prefer not to have `TimeoutType` in the message.

**Review by alexshtin — CHANGES_REQUESTED** [review]
.

File: service/history/timerQueueActiveTaskExecutor.go (hunk 2 above)
Comment (feedmeapples) [reply service/history/timerQueueActiveTaskExecutor.go:222]: good point

**Review by alexshtin — APPROVED** [review]

**Pull request merged**